### PR TITLE
adding swagger documentation

### PR DESCRIPTION
**Pull request by spiderxm**
Added Swagger and Redoc documentation for rest apis.

### CODE REVIEW

**Review by mlodic — CHANGES_REQUESTED** [review]
Thank you for your PR!

However it is really important that you read and follow the contributions guidelines.

Also please do not delete the PR template. You should compile it. Otherwise your contributions won't be considered. Everything you need to know is written there.

**Comment by Eshaan7** [comment]
We need to make this autogenerate for every commit with rest of the docs and host them publicly like: https://intelowl.readthedocs.io/api-docs. 

How can we achieve that?

**Comment by spiderxm** [comment]
We can use the command
> python manage.py python manage.py generateschema 

This will generate open api definitions in yaml.
That can be used to make swagger documentation and serve them via different application.

**Comment by Eshaan7** [comment]
> This will generate open api definitions in yaml.
> That can be used to make swagger documentation and serve them via different application.

Okay so what we need to do is create a GH action (or readthedocs webhook) which builds the `.yml` schema and puts it into `/docs/source/` which then gets rendered using [`sphinx openapi extension`](https://github.com/sphinx-contrib/openapi) ? The catch here is that this action should take place before readthedocs build gets triggered for which you should find something in readthedocs documentation.

File: api_app/urls.py [hunk 1]
```diff
@@ -33,3 +56,4 @@
     # Viewsets
     path(r"", include(router.urls)),
 ]
+
```
Comment (Eshaan7) [inline api_app/urls.py:59]: please fix codefactor issue

File: api_app/urls.py [hunk 2]
```diff
@@ -13,14 +17,33 @@
...
 from .auth import LoginView, LogoutView
 
-
 # Routers provide an easy way of automatically determining the URL conf.
 router = routers.DefaultRouter(trailing_slash=False)
 router.register(r"tags", TagViewSet)
 router.register(r"jobs", JobViewSet)
 
+# Documentation Settings
+schema_view = get_schema_view(
```
Comment (Eshaan7) [inline api_app/urls.py:26]: instead of adding this here, we should add a new file called `schemas.py` and add all this docs related code there.

**Comment by spiderxm** [comment]
Will look into it

**Comment by Eshaan7** [comment]
https://github.com/sphinx-contrib/redoc/issues/15

**Pull request closed without merging**